### PR TITLE
Openthread v1.0

### PR DESCRIPTION
**Pull request by ddreyer**
Debug code to collect OpenThread data

### CODE REVIEW

File: src/core/mac/mac.cpp [hunk 1]
```diff
@@ -1091,7 +1091,7 @@ otError Mac::RadioTransmit(Frame *aSendFrame)
     SuccessOrExit(error = otPlatRadioTransmit(&GetInstance(), static_cast<otRadioFrame *>(aSendFrame)));
 
 exit:
-
+
```
Comment (Hyungsin) [inline src/core/mac/mac.cpp]: Please remove these redundant changes

File: src/core/thread/mle_router.cpp [hunk 2]
```diff
@@ -2145,838 +2172,7 @@ otError MleRouter::SendParentResponse(Child *aChild, const ChallengeTlv &aChalle
     LogMleMessage("Delay P-Resp", destination);
 #if ENABLE_DEBUG
     otPlatLog(OT_LOG_LEVEL_INFO, OT_LOG_REGION_MLE, "[OT-MLERouter]: Delay P-Resp\n");
-#endif
```
Comment (Hyungsin) [inline src/core/thread/mle_router.cpp]: Why is it happen? So many lines are removed?

File: src/core/common/debug.hpp [hunk 3]
```diff
@@ -79,5 +79,18 @@
...
+extern uint32_t joiningMsgCnt;
+extern uint32_t routingMsgCnt;
+extern uint32_t linkMsgCnt;
+extern uint32_t controlMsgCnt;
+extern uint32_t packetSuccessCnt;
+extern uint32_t packetFailCnt;
+extern uint32_t packetBusyChannelCnt;
+extern uint32_t broadcastCnt;
+extern uint16_t myRloc;
+extern uint16_t nextHopRloc;
```
Comment (Hyungsin) [inline src/core/common/debug.hpp]: Where are they initialized to zero? 
and please put a description of each variable.

**Comment by Hyungsin** [comment]
Ok I put some comments. Please figure these out.

**Review by Hyungsin — CHANGES_REQUESTED** [review]
Ok I put some comments. Please figure these out.
Also please make this PR as a single commit.

File: src/core/common/debug.hpp (hunk 3 above)
Comment (ddreyer) [reply src/core/common/debug.hpp]: They are initialized to zero in the main function

File: src/core/mac/mac.cpp [hunk 4]
```diff
@@ -1539,7 +1544,7 @@ extern "C" void otPlatRadioReceiveDone(otInstance *aInstance, otRadioFrame *aFra
     }
     else
 #endif // OPENTHREAD_ENABLE_RAW_LINK_API
-    {
+    {
```
Comment (Hyungsin) [inline src/core/mac/mac.cpp]: please remove this redundant space

File: src/core/thread/mesh_forwarder.cpp [hunk 5]
```diff
@@ -2425,7 +2425,7 @@ void MeshForwarder::HandleDataRequest(const Mac::Address &aMacSource, const otTh
     otLogInfoMac(GetInstance(), "Rx data poll, src:0x%04x, qed_msgs:%d, rss:%d", child->GetRloc16(), indirectMsgCount,
                  aLinkInfo.mRss);
 #if ENABLE_DEBUG
-    otPlatLog(OT_LOG_LEVEL_INFO, OT_LOG_REGION_IP6, "[OT-MF]: Rx Data Poll, src:0x%04x, qed_msgs:%d\n", child->GetRloc16(), indirectMsgCount);
+    otPlatLog(OT_LOG_LEVEL_INFO, OT_LOG_REGION_IP6, "[OT-MeshForward]: Rx Data Poll, src:0x%04x, qed_msgs:%d\n", child->GetRloc16(), indirectMsgCount);
```
Comment (Hyungsin) [inline src/core/thread/mesh_forwarder.cpp]: please use MF instead of MeshForward

File: src/core/thread/mle_router.cpp [hunk 6]
```diff
@@ -1717,7 +1720,7 @@ void MleRouter::UpdateRoutes(const RouteTlv &aRoute, uint8_t aRouterId)
 
                 update |= mRouters[i].GetNextHop() != oldNextHop;
             }
-
+
```
Comment (Hyungsin) [inline src/core/thread/mle_router.cpp]: Please remove this redundant change

File: src/core/thread/mle_router.cpp [hunk 7]
```diff
@@ -1747,6 +1750,13 @@ void MleRouter::UpdateRoutes(const RouteTlv &aRoute, uint8_t aRouterId)
     }
     otPlatLog(OT_LOG_LEVEL_INFO, OT_LOG_REGION_MLE, "\n\n---- [OT-RT] ----\n");
     otPlatLog(OT_LOG_LEVEL_INFO, OT_LOG_REGION_MLE, " ID NH RC LC NHLC\n");//InLinkQ  OutLinkQ\n");    
+
+    // resolve border router EID to RLOC16
+    ot::Ip6::Address borderIP;
+    otIp6AddressFromString("fdde:ad00:beef:0000:c684:4ab6:ac8f:9fe5", &borderIP);
+    otShortAddress borderRloc16;
+    GetNetif().GetAddressResolver().Resolve(borderIP, borderRloc16);
+    otPlatLog(OT_LOG_LEVEL_INFO, OT_LOG_REGION_MLE, "BR RLOC16: %d\n", borderRloc16);
```
Comment (Hyungsin) [inline src/core/thread/mle_router.cpp]: please remove the redundant space at the end of this line

**Review by Hyungsin — CHANGES_REQUESTED** [review]
I added some comments.
Also, please remove all redundant spaces at the end of each line.

**Comment by ddreyer** [comment]
I have fixed the redundant spaces

**Pull request merged**